### DISCUSSION TITLE
How to best connect multiple 1-Wire sensors into a single SH-ESP32 or HALMET board

### DISCUSSION THREAD

**Discussion by mairas** [hatlabs/discussions#82] [General]
Someone asked me over email how to best connect multiple 1-Wire sensors into a single SH-ESP32 or HALMET board and I thought I'd expand the answer a bit and publish it here for wider consumption.

With one to three sensors, it's feasible to add panel connectors straight to the enclosure and connect them together in a pigtail, as described in my 1-Wire tutorial: https://docs.hatlabs.fi/sh-esp32/pages/tutorials/onewire-temperature/. However, if you have more sensors or are using HALMET and need to have other connectors on the enclosure as well, it's better to connect the one-wire cables together outside of the device enclosure.

Use a junction box to protect the connection. I have found standard electrical junction boxes to be quite suitable. They're very inexpensive, have integrated grommets, and you can get them with decent IP ratings. For the actual connection, I don't have a perfect solution, but some alternatives. It's not a high-speed connector, though, so as long as you can get the individual wires together in a reliable way, anything goes. Well, almost.

If it's fine for the connection to be permanent, a simple pigtail splice with solder and heat shrink would do. If you want to be able to disconnect and replace individual sensors (which is not a bad idea), I'd probably use Wago-style lever terminal blocks. They're a bit bulky but are still rated for really thin wires, too. They come up with up to 5 terminals, but if you want to connect more cables or want to use a bit smaller terminal blocks, you can twist a 2-3 wires together and clamp them under the lever and it works fine.
Finally, I'd use a zip-tie as a strain-relief for the connector, and depending on the installation location, I might smear a bit of dielectric grease inside the connector to protect the bare wires from corrosion.

**Comment by big-col** [comment]
I use Wago connectors for my 1-wire sensors, but I only have 5 currently so quite easy
